coriolis/tests/scheduler/rpc/test_client.py

```diff
@@ -0,0 +1,242 @@
+            mock.sentinel.ctxt, provider_requirements=None, region_sets=None,
+            enabled=True)
+        self.assertEqual(result, mock_service)
+
+    @mock.patch.object(client.SchedulerClient, 'get_worker_service_for_specs')
+    @mock.patch.object(tasks_factory, 'get_task_runner_class')
+    def test_get_worker_service_for_task_different_platforms(
+            self, mock_get_task_runner_class, get_worker_service_for_specs):
+        for platform in [constants.TASK_PLATFORM_SOURCE,
+                         constants.TASK_PLATFORM_DESTINATION,
+                         constants.TASK_PLATFORM_BILATERAL]:
+            mock_get_task_runner_class.return_value.get_required_platform.\
+                return_value = platform
+            mock_get_task_runner_class.return_value.\
+                get_required_provider_types.return_value = {
+                    constants.PROVIDER_PLATFORM_SOURCE: 'provider_type'}
+            get_worker_service_for_specs.return_value = {'id': 'test_id'}
+
+            result = self.client.get_worker_service_for_task(
+                mock.sentinel.ctxt, self.task, self.origin_endpoint,
+                self.destination_endpoint, retry_period=0)
+
+            self.assertEqual(result, {'id': 'test_id'})
+            self.assertIsInstance(result, dict)
+
+    @mock.patch.object(client.SchedulerClient, 'get_worker_service_for_specs')
+    @mock.patch.object(tasks_factory, 'get_task_runner_class')
+    def test_get_worker_service_for_task_retry(
+            self, mock_get_task_runner_class, mock_get_worker_service):
+        mock_get_task_runner_class.return_value.get_required_platform.\
+            return_value = constants.TASK_PLATFORM_SOURCE
+        mock_get_task_runner_class.return_value.get_required_provider_types.\
+            return_value = {constants.PROVIDER_PLATFORM_DESTINATION:
+                            'provider_type'}
+        mock_get_worker_service.side_effect = [Exception(), {'id': 'test_id'}]
+
+        with self.assertLogs('coriolis.scheduler.rpc.client',
+                             level=logging.WARN):
+            self.client.get_worker_service_for_task(
+                mock.sentinel.ctxt, self.task, self.origin_endpoint,
+                self.destination_endpoint, retry_period=0)
+
+    @mock.patch.object(client.SchedulerClient, 'get_worker_service_for_specs')
+    @mock.patch.object(tasks_factory, 'get_task_runner_class')
+    def test_get_worker_service_for_task_no_suitable_worker(
+            self, mock_get_task_runner_class, mock_get_worker_service):
+        mock_get_task_runner_class.return_value.get_required_platform.\
+            return_value = constants.TASK_PLATFORM_SOURCE
+        mock_get_task_runner_class.return_value.get_required_provider_types.\
+            return_value = {constants.PROVIDER_PLATFORM_SOURCE: 'type'}
+        mock_get_worker_service.\
+            side_effect = [exception.NoSuitableWorkerServiceError()]
```

Comment:
NIT: We usually prefer this code style instead:
```suggestion
        mock_get_worker_service.side_effect = [
            exception.NoSuitableWorkerServiceError()]
```

Please apply this style from now on